benchexec/tools/abc.py

```diff
@@ -24,7 +25,33 @@ def name(self):
         return "ABC"
 
     def cmdline(self, executable, options, task, rlimits):
-        return [executable] + options + [task.single_input_file]
+        return (
+            [executable]
+            + ["-c", "&r {}; &put".format(task.single_input_file)]
```

Comment:
Is this general enough for all commands in ABC?
Remember that this tool-info module might be used for other applications besides verification.

Reply:
I changed the read method because the old ABC read method cannot process uninitialized registers correctly (see e1e2b06).

The circuit is stored as a GIA (the newer AIG representation) after `&read` (`&r`, the new read method).
Then `&put` create a copy the circuit and store it as an AIG (the old AIG representation).
Any following command can be executed no matter it needs a GIA or AIG.

However, this new `&read` can only take an AIGER file as input, whereas the old ABC `read` can consume other formats such as BLIF.

Reply:
If we want to support other file formats, we can also add a switch here based on the file extension of the input.

Reply:
Yes, we want to support other file formats, too.
Could you add the switch and some documentation here?

Reply:
Please have a look at c2ccbdd.